### ISSUE TITLE
VS Code Insider: ver 1.11.0 issues with extensions can't be uninstalled.

### ISSUE THREAD

**Issue by MaximoTrinidad** [Microsoft/vscode#21983]
- VSCode Version: Code - Insiders 1.11.0-insider 
- OS Version: Linux Ubuntu 16.04.1
- Extensions: 
|Extension|Author|Version|
|---|---|---|
|html-snippets|abusaidm|0.1.0|
|python|donjayamanne|0.5.9|
|code-runner|formulahendry|0.6.12|
|csharpextensions|jchannon|1.3.0|
|mssql|ms-mssql|0.3.0|
|csharp|ms-vscode|1.7.0|
|PowerShell|ms-vscode|0.9.0|
|vscode-icons|robertohuertasm|7.3.0|
|python|tht13|0.2.3|
|python-extended-snippets|tushortz|0.0.1|;

---

In reference to an issue posted "MSSQL: SQL Tools Service not starting on latest VS Code Insider #718", I discover a possible bug with all extension not able to be uninstalled.

Posted on MSSQL issue:

I had mssql extension (version 0.3.0) working on VS Code Insider 1.10.0 on my Linux machine working fine. But, I proceeded to update to VS Code to version 1.11.0 and now I'm getting an error: "Unable to execute the command while the extension is initializing".

And, it's worse! I can't uninstalled any of my existing extensions. Clicking on the "uninstall" won't do anything. I can only to disable/enable but it won't clear the issue.

The SqlToolsService is hung up trying to initialize the service. No debug error message is generated.
I'm on Linux Ubuntu 16.04.1, VS Code 1.11.0.

Now, the Windows version VS Code ver. 1.11.0 and the extensions "uninstall" works fine.

**Comment by sandy081** [comment]
@MaximoTrinidad Above description is not helping to reproduce/diagnose the issue. Installing and uninstalling mssql or other extensions is working for me. 

I am also not sure where is the error `Unable to execute the command while the extension is initializing` being coming from. Is it the mssql extension?

Can you please provide the steps to reproduce the issue?

**Comment by MaximoTrinidad** [comment]
My apologies for the confusion!  As, I was testing the mssql extension giving the error I posted in both their GitHub issue and I copy/paste the same text.  But, the issue I'm relaying here is about any extensions.

In order to **reproduce this issue**.   
1. Go to your list of installed extension.
2. On any of the extension(s), click "uninstall".

**Expected Behavior:**  Proceed to uninstall the extension.

**Actual Behavior:**  It won't uninstall the extension.

Then again, this is only on my Linux **Ubuntu 16.04.1** OS.  
The Windows VS Code Insider works fine!

**Comment by MaximoTrinidad** [comment]
I downloaded the latest version now and I still can't "uninstall" any extensions.
** Linux Ubuntu 16.04.1**  (Windows is fine!)

- VSCode Version: Code - Insiders 1.11.0-insider (a3b1076f079c150d921a7b91d91c6783fa04450a, 2017-03-06T07:25:37.413Z)
- OS Version: Linux x64 4.4.0-65-generic
- Extensions: 

|Extension|Author|Version|
|---|---|---|
|python|donjayamanne|0.5.9|
|code-runner|formulahendry|0.6.13|
|mssql|ms-mssql|0.3.0|
|PowerShell|ms-vscode|0.9.0|
|vscode-icons|robertohuertasm|7.3.0|
|python|tht13|0.2.3|
|python-extended-snippets|tushortz|0.0.1|;

![vscodelinux_uninstallissue](https://cloud.githubusercontent.com/assets/19892435/23637879/2cc1db80-02ac-11e7-933b-7b52aa43504c.png)

**Comment by sandy081** [comment]
@MaximoTrinidad Thanks for the information. I tried on Linux Ubuntu 14.04 with the same set of extensions you have and everything (Uninstall) works for me.

Can you do following

- Open Developer tools using command `Toggle Developer tools` and see if there are any errors or anything suspecting while uninstalling?
- Is it possible to provide a video of what you are doing?

Thanks

**Comment by MaximoTrinidad** [comment]
Hi @sandy081,

I use the "Toggle Developer tools" and try to 'uninstall' any of the extensions. It is showing errors when I click on console.
  
Sorry! But I don't have any video recording app on my Linux machine.

Also, after opening VS Code Insiders another "Available Update".  Let me complete this update and try again,  

If there are any errors, I will provide the screen shot.

:)

**Comment by MaximoTrinidad** [comment]
@sandy081,

Today I dDownload file: sudo dpkg -i code-insiders_1.11.0-1488871392_amd64.deb

- VSCode Version: Code - Insiders 1.11.0-insider (5e9357bd5111eacf2b8fb39455c8afb5a42d632f, 2017-03-07T07:28:38.651Z)
- OS Version: Linux x64 4.4.0-65-generic
- Extensions: 

|Extension|Author|Version|
|---|---|---|
|python|donjayamanne|0.5.9|
|code-runner|formulahendry|0.6.13|
|mssql|ms-mssql|0.3.0|
|PowerShell|ms-vscode|0.9.0|
|vscode-icons|robertohuertasm|7.3.0|
|python|tht13|0.2.3|
|python-extended-snippets|tushortz|0.0.1|;

(See errors on attached image)

I still can't "uninstall" any of the extensions I have previously installed.

:(

![screenshot from 2017-03-07 11-14-25](https://cloud.githubusercontent.com/assets/19892435/23667162/03bcc6ee-032b-11e7-8eb0-d589215d85f6.png)

**Comment by sandy081** [comment]
@MaximoTrinidad May I know what you see when you open the dropdown on the settings gear icon and click uninstall?

Also can you disable all extensions using the command in Command Pallette - `Disable all Installed Extensions`. Restart VS Code and install an extension (say TSLint) and try to uninstall.

Regarding the errors, can you open the message and provide the information

**Comment by MaximoTrinidad** [comment]
@sandy081, 

I will have try tonight when I get home.  By the way, I can disable/enable but I will try your steps later and collect more information.

:)

**Comment by MaximoTrinidad** [comment]
@sandy081,

I tried your instructions and still can't uninstall any of the existing installed extensions.  I can add more extensions. 

But, guess what???  I decided to create an Ubuntu 16.04.1 VM and installed the latest VS Code Insider. It is working as expected.

It seems , my VS Code Insider on my main Ubuntu system in not working correctly.  

I guess I need to know how to do a clean installation. How to manually remove all VS Code Insider components. 

Sorry for the confusion!!
:)
